homeassistant/components/home_connect/__init__.py

```diff
@@ -110,8 +110,7 @@ def _get_appliance_by_device_id(
             device = dev_dict[CONF_DEVICE]
             if device.device_id == device_id:
                 return device.appliance
-    _LOGGER.error("Appliance for device id %s not found", device_id)
-    return None
+    raise ValueError(f"Appliance for device id {device_id} not found")
```

Comment:
Please update the return value typing of the function signature. It doesn't return None anymore.

Reply:
Visual Studio Code said the return statement was unreachable code.

Reply:
I'm referring to the return value type annotation on line 106.